src/saturn_engine/worker/job/__init__.py

```diff
@@ -0,0 +1,53 @@
+import dataclasses
+from typing import Optional
+
+from saturn_engine.utils.log import getLogger
+from saturn_engine.utils.options import OptionsSchema
+
+from ..inventories import Inventory
+from ..queues import Publisher
+
+
+class JobStore(OptionsSchema):
+    async def load_cursor(self) -> Optional[str]:
+        pass
+
+    async def save_cursor(self, *, after: str) -> None:
+        pass
+
+
+@dataclasses.dataclass
+class JobItem:
+    id: str
+    data: object
+
+
+class Job:
+    after: Optional[str]
```

Comment:
Any reason you don't type those in the init?

Now everytime I read a constructor I also have to go look at the class def for **part** of the typing definitions.

Reply:
If we define everything in the class body then there are still "surprise" types.

Let me demonstrate:

```

class A:
   # Reviewer reads here. He thinks is all OK.
    a: int
    b: float


    def __init__(self) -> None:
        a = 1
        b = 2
        # Reviewer does not notice that c was missing from the class body. c is now typed as an int
        # but if the reviewer had noticed maybe he would have commented that it should be an Optiona[int].
        c = 3
```

Reply:
Other possible bug:

```
class A:
    a: int

    def __init__(self):
        pass

a = A()
print(a.a) # boom
```